torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py

```diff
@@ -178,6 +179,11 @@ def detach_and_copy_item_memo(t):
 
         hook.remove()  # type: ignore[possibly-undefined]
 
+    if not torch._dynamo.config.skip_fsdp_hooks:
+        fw_module.graph = fx_passes.raise_fsdp2_backward_all_gather_ops_if_applicable(
+            fw_module.graph
+        )
```

Comment:
We intentionally run this pass at the end of the aot_inference graph generation path (instead of running in Inductor post-grad), so that there is no use-before-write issue in aot_inference graph. This will make `aot_eager` backend work which makes it better for production debugging.

Reply:
I'm still trying to understand why this pass is necessary 🤔. If I have some code in the forward like this:
```
def f(x):
    y = op1(x)
    y.register_hook(hook)
    z = op2(y)
    return z
```

Then I should end up with a compiled autograd graph that looks something like this:
```
def backwad_graph(tangent1):
    tangent2 = op2_backward(tangent1)
    tangent2 = hook(tangent2)
    tangent3 = op1_backward(tangent2)
```

and the `hook` will turn into the `parameter.resize_(full_size)`, happening (importantly, before `op1_backward` runs).

are you seeing an example where the resizes are getting moved to be after the corresponding backward node?